Source/components/Text.js

```diff
@@ -188,81 +132,99 @@ export class Text extends Layer {
 
     return fragments
   }
+}
+
+Text.type = Types.Text
+Text[DefinedPropertiesKey] = { ...Layer[DefinedPropertiesKey] }
+Factory.registerClass(Text, MSTextLayer)
 
+Text.define('text', {
+  get() {
+    return String(this._object.stringValue())
+  },
   /**
-   * Set whether to use constant baseline line spacing mode.
+   * Set the text of the layer.
+   * If the layer hasn't explicitly been given a name, this will also change
+   * the layer's name to the text value.
    *
-   * @param {bool} value If true, we use constant baseline spacing mode. This is the default for new text layers in Sketch. If false, we use the legacy line spacing mode.
+   * @param {string} value The text to use.
    */
-  set useConstantBaselines(value) {
-    const lineSpacingBehaviour = value
-      ? TextLineSpacingBehaviour.constantBaseline
-      : TextLineSpacingBehaviour.variable
-    const textLayer = this.sketchObject
-    const initialBaselineOffset = textLayer.firstBaselineOffset()
-    textLayer.lineSpacingBehaviour = lineSpacingBehaviour
-    const baselineOffset = textLayer.firstBaselineOffset()
-    const rect = this.frame
-    rect.y -= baselineOffset - initialBaselineOffset
-    this.frame = rect
-  }
+  set(value) {
+    const object = this._object
+    object.stringValue = value
+    if (!object.nameIsFixed()) {
```

Comment:
Don't we deal with this on Sketch's side? Even if you want to do this manually here, either use `object.updateNameFromStorage()` or check the code there, because the logic we use to set the name is not as simple as `name = value`.

Reply:
that was the existing code